### DISCUSSION TITLE
Multilingual support

### DISCUSSION THREAD

**Discussion by nikos4github** [Azure/api-management-developer-portal#2538] [Feature request]
Can we have a multilingual dev portal? Is this supported?

**Comment by mikebudzynski** [comment]
@nikos4github 

Currently localization is not supported in the developer portal.

In general, there are two areas that fall under localization:

1. Localization of static content (for example, text, headers, titles)
2. Localization of dynamic data (for example, API descriptions)

We don't have timelines yet.

**Comment by fuchsdaniel** [comment]
Guys,

Is there any update on this feature request?
regards,

Daniel

**Comment by mikebudzynski** [comment]
No updates; likely we won't be able to work on it within the next 8 months.

**Comment by mapoitras** [comment]
+1 request for this important feature.

**Comment by ncroad** [comment]
Is it possible to get an update on this feature request? A client needs parts of this feature and I would like to be able to set some expectations.

Thanks.

**Comment by freeriderND** [comment]
Is there any update on this feature request?

**Comment by mikebudzynski** [comment]
No update at the moment. We don't plan to work on this feature in the next 6 months.

**Comment by yanickgirouard** [comment]
This is a rather major feature to have if you plan on having any non-US customers using it. It will be a show stopper for most French companies that need the portal to be in French (i.e. in Quebec). Is there no way to localize the new developer portal at the moment?

**Comment by mikebudzynski** [comment]
> Is there no way to localize the new developer portal at the moment?

The only way to localize hardcoded English text in the widgets is to modify the portal's code for translations and then self-host it.

**Comment by robarui** [comment]
Any update on this?

**Comment by robarui** [comment]
> Any update on this?

@mikebudzynski

**Comment by mikebudzynski** [comment]
@robarui, this feature is on our roadmap, but we won't be able to work on it in the next 6 months.

**Comment by mborejdo** [comment]
is there any update on this? any eta?

**Comment by cburatto** [comment]
Ping. That's a valuable feature.

**Comment by basantos96** [comment]
Hi, Any updates on this?

**Comment by MatiasUnsvag** [comment]
Same, would like to have this feature.

**Comment by alexbenitez** [comment]
It's been more than 3 years... any hint that this will ever be implemented? Looking for a bilingual portal (English/French users). All pre-built widget are English only, unfortunately.

**Comment by yanickgirouard** [comment]
It's been almost 6 years later and still no multilingual support built-in... This is kind of sad and almost insulting to all non-English customers. Self-hosting the portal is tedious, complex and simply out of reach for many companies, and doesn't even completely solve the localization issue. Would it be possible to bump this up in priority on your roadmap? The [documentation](https://learn.microsoft.com/en-us/azure/api-management/developer-portal-self-host) might be available, but it doesn't cover for all the work involved in maintaining the self-hosted portal code up to date and current with your latest changes.
